### ISSUE TITLE
First payment fails with FAILURE_REASON_TIMEOUT, subsequent succeds

### ISSUE THREAD

**Issue by Overtorment** [lightningnetwork/lnd#4267]
### Background

Same bolt11 from tippin.me. Trying to send:

```
overtorment@lnd2:~/go/src/github.com$ lncli sendpayment --pay_req   lnbc1p0t2a8lpp5997jfdgkd3ngklg4h8z7ll9tg3f3vyjcrj5emzzmf75hk9njmppqdpu235hqurfdcsx7an9wf6x7undv4h8ggpgw35hqurfdchx6eff9p6nzvfc8q5scqzpgxqyz5vqsp5w83nulzj2kwh8p7e0k0w0zgljpusyr4y8sll972rlng9quqnmwqq9qy9qsqrr225tzh94ukj0yrer93rw3tcnu5kurhvmkmex5h5ccyalt8ah39la6a55dda95rwrs5pld79zhm3elvkfx7nvaa64q6qr8hfznkrvcpwjkstq  --amt 1   --force
+------------+--------------+--------------+--------------+-----+----------+--------------------+-----------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT           | ROUTE     |
+------------+--------------+--------------+--------------+-----+----------+--------------------+-----------+
| IN_FLIGHT  |        0.034 |            - | 1            | 0   |   629556 | 691787427480600576 | tippin.me |
+------------+--------------+--------------+--------------+-----+----------+--------------------+-----------+
+-------------------------------+--------------+--------------+--------------+-----+----------+--------------------+-----------+
| HTLC_STATE                    | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT           | ROUTE     |
+-------------------------------+--------------+--------------+--------------+-----+----------+--------------------+-----------+
| TEMPORARY_CHANNEL_FAILURE @ 0 |        0.034 |       60.119 | 1            | 0   |   629556 | 691787427480600576 | tippin.me |
+-------------------------------+--------------+--------------+--------------+-----+----------+--------------------+-----------+
Amount + fee:   0 + 0 sat
Payment hash:   297d24b5166c668b7d15b9c5effcab44531612581ca99d885b4fa97b1672d842
Payment status: FAILED, reason: FAILURE_REASON_TIMEOUT
[lncli] FAILED

```

while tailgrepping:

```
overtorment@lnd2:~/go/src/github.com$ pm2 logs 2 | grep 03c2abfa93eace
2|run-lnd  | 2020-05-08 15:04:44.300 [DBG] PEER: Sending UpdateAddHTLC(chan_id=0030bcccd6cea535ab0481842ef80bb5fb7927d76f876b57b5ac89c4b7c8cadf, id=76, amt=1000 mSAT, expiry=629556, hash=297d24b5166c668b7d15b9c5effcab44531612581ca99d885b4fa97b1672d842) to 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:48.836 [DBG] PEER: Sending Ping to 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:59.261 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626334599913406465, mflags=00000001, cflags=00000000, update_time=2020-05-07 19:21:26 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:59.266 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626334599913406465, mflags=00000001, cflags=00000001, update_time=2020-05-07 17:03:30 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:59.268 [DBG] PEER: Received ChannelAnnouncement(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626334599913603072) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:59.277 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626334599913603072, mflags=00000001, cflags=00000010, update_time=2020-05-07 20:13:26 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:59.281 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626334599913603072, mflags=00000001, cflags=00000001, update_time=2020-03-08 00:08:07 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:59.286 [DBG] PEER: Received ChannelAnnouncement(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626340097409482752) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:04:59.292 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626340097409482752, mflags=00000001, cflags=00000010, update_time=2020-05-08 07:55:45 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:05:48.836 [DBG] PEER: Sending Ping to 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902





```

same bolt11, take two:

```
overtorment@lnd2:~/go/src/github.com$ lncli sendpayment --pay_req   lnbc1p0t2a8lpp5997jfdgkd3ngklg4h8z7ll9tg3f3vyjcrj5emzzmf75hk9njmppqdpu235hqurfdcsx7an9wf6x7undv4h8ggpgw35hqurfdchx6eff9p6nzvfc8q5scqzpgxqyz5vqsp5w83nulzj2kwh8p7e0k0w0zgljpusyr4y8sll972rlng9quqnmwqq9qy9qsqrr225tzh94ukj0yrer93rw3tcnu5kurhvmkmex5h5ccyalt8ah39la6a55dda95rwrs5pld79zhm3elvkfx7nvaa64q6qr8hfznkrvcpwjkstq  --amt 1   --force
+------------+--------------+--------------+--------------+-------+----------+--------------------+--------------------------------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE   | TIMELOCK | CHAN_OUT           | ROUTE                          |
+------------+--------------+--------------+--------------+-------+----------+--------------------+--------------------------------+
| SUCCEEDED  |        0.057 |        1.602 | 1            | 0.001 |   629700 | 691773133799555072 | ln1.satoshilabs.com->tippin.me |
+------------+--------------+--------------+--------------+-------+----------+--------------------+--------------------------------+
Amount + fee:   1 + 0.001 sat
Payment hash:   297d24b5166c668b7d15b9c5effcab44531612581ca99d885b4fa97b1672d842
Payment status: SUCCEEDED, preimage: 338e97ea454daf3712216c02eda1b821de5a817c20c4b2772e82517cf25f9a88
```

while tailgrepping:

```






2|run-lnd  | 2020-05-08 15:06:01.176 [DBG] CRTR: Reporting pair success to Mission Control: pair=0279c22ed7a068d10dc1a38ae66d2d6461e269226c60258c021b1ddcdfe4b00bc4 -> 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda, amt=1000 mSAT
2|run-lnd  | 2020-05-08 15:06:01.176 [DBG] CRTR: Setting 0279c22ed7a068d10dc1a38ae66d2d6461e269226c60258c021b1ddcdfe4b00bc4->03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda range to [100000 mSAT-0 mSAT]
2|run-lnd  | 2020-05-08 15:06:39.124 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626340097409482752, mflags=00000001, cflags=00000001, update_time=2020-04-30 14:00:35 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:06:48.836 [DBG] PEER: Sending Ping to 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:07:48.836 [DBG] PEER: Sending Ping to 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:07:58.450 [DBG] PEER: Received ChannelAnnouncement(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626345594971160576) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:07:58.454 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626345594971160576, mflags=00000001, cflags=00000000, update_time=2020-05-08 12:13:14 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
2|run-lnd  | 2020-05-08 15:07:58.461 [DBG] PEER: Received ChannelUpdate(chain_hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, short_chan_id=626345594971160576, mflags=00000001, cflags=00000001, update_time=2020-05-06 11:52:28 +0000 UTC) from 03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda@157.245.68.47:45902
```

### Your environment

* version of `lnd` 0.10
* which operating system (`uname -a` on *Nix) Ubuntu 18
* version of `btcd`, `bitcoind`, or other backend neutrino
* any other relevant environment details

### Steps to reproduce

see above

### Expected behaviour

payments succeed first time. direct channel with tippin.me, active, present in listpeers

### Actual behaviour

1st attempt fails

**Comment by cfromknecht** [comment]
Appears the first HTLC got stuck locally and then failed back after 1 min. Possibly fixed by #4183?

**Comment by juscamarena** [comment]
I think this makes the case for having unique payment ids like eclair does, otherwise if you retry a payment how can you differentiate the first attempt and the 2nd attempt?

**Comment by Roasbeef** [comment]
Could've also been the case that the link flapped right after it was handed off? Then eventually got cancelled back by the new mailbox logic.

**Comment by cfromknecht** [comment]
> Could've also been the case that the link flapped right after it was handed off? Then eventually got cancelled back by the new mailbox logic.

Mhm definitely possible, seems related because of the 1 minute timeout. I suspect #4183 would resolve this because locally-initiated HTLCs will skip the mailboxes entirely.

**Comment by Roasbeef** [comment]
We believe this was resolved by #4183.
